.github/workflows/digest-updater.yaml

```diff
@@ -0,0 +1,87 @@
+
+    steps:
+      - name: Configure Git
+        run: |
+         git config --global user.email "github-actions[bot]@users.noreply.github.com"
+         git config --global user.name "GitHub Actions"
+
+      # Get the latest weekly build commit hash: https://github.com/opendatahub-io/notebooks/commits/2023a
+      - name: Checkout upstream notebooks repo
+        uses: actions/checkout@v3
+        with:
+         repository: opendatahub-io/notebooks.git
+         ref: 2023a
+      - name: Retrive latest weekly commit hash from the release branch
+        id: hash
+        shell: bash
+        run: |
+          echo "HASH=$(git rev-parse --short HEAD)" >> ${GITHUB_OUTPUT}
+
+      - name: Checkout main from the odh-manifest repo
+        uses: actions/checkout@v3
+        with:
+          ref: main
+      # Get image name from: .spec.tags[0].from.name
+      - name: Fetch the name of the image from Image Stream (Step 1)
+        shell: bash
+        id: img_name
+        run: |
+         echo "img_name=$(yq '.spec.tags[0].from.name' ${{ matrix.files }} | sed 's/\"/\//g')" >> ${GITHUB_OUTPUT}
+
+      # Checkout the release branch to apply the updates
+      - name: Checkout release branch
+        uses: actions/checkout@v3
+        with:
+          ref:  ${{ inputs.branch }}
+
+      - name: Fetch digest, and update the imagestream file (Step 2)
+        env:
+          IMG_NAME: ${{ steps.img_name.outputs.img_name }}
+        run: |
+          src_tag=$(echo "${{ env.IMG_NAME }}" | sed 's/-weekly.*//' | cut -d':' -f2 | tr -d '"')
+          regex="$src_tag-\d+-${{ steps.hash.outputs.HASH }}"
+          echo $regex
+
+          # Get the latest tag: output ex. latest_tag=jupyter-minimal-ubi9-python-3.9-2023a-20230403-cf92364
+          latest_tag=$(skopeo inspect docker://${{ env.IMG_NAME }} | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
+          echo $latest_tag
+
+          #Fetch the latest tag in Digest format: ex.digest=quay.io/opendatahub/workbench-images@sha256:xxx...xxx
+          digest=$(skopeo inspect docker://quay.io/opendatahub/workbench-images:$latest_tag | jq .Digest | sed 's/sha256/quay.io\/opendatahub\/workbench-images@sha256/g')
+          digest=$(echo "$digest" | tr -d '"' | tr -d '\\')
+          echo $digest
+
+          # Update the .spec.tags[0].from.name field on the release image stream with the digest outputed digest
+          yq eval '.spec.tags[0].from.name = "'$digest'"' -i ${{ matrix.files }}
+
+          # Fetch, pull, add, and commit the changes into the release branch
+          git fetch origin ${{ inputs.branch }} && git pull origin ${{ inputs.branch}} && git add ${{ matrix.files }} && git commit -m "Update file ${{ matrix.files }} via digest-updater GitHub action" && git push origin ${{ inputs.branch }}
```

Comment:
Can we change this a create a Pull Request instead of merging directly to the release branch as a final check to make sure that there is no error in this action

Reply:
Right, I updated the workflow to open a pull request instead. 
![image](https://user-images.githubusercontent.com/42587738/230069159-b6cc1796-063a-4a60-999d-ad245ba87a1f.png)